keylime/signing.py

```diff
@@ -0,0 +1,77 @@
+'''
+SPDX-License-Identifier: Apache-2.0
+Copyright 2017 Massachusetts Institute of Technology.
+'''
+
+import gnupg
+import os
+
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.primitives.serialization import load_pem_public_key
+
+from keylime import keylime_logging
+
+logger = keylime_logging.init_logging('signing')
+
+def verify_signature_from_disk(key_file, filename, sig_file, file_description):
+    """
+       Read signing materials from disk and verify.
+    """
+
+    with open(key_file, 'rb') as key_f:
+        key = key_f.read()
+    with open(sig_file, 'rb') as sig_f:
+        sig = sig_f.read()
+    with open(filename, 'rb') as file_f:
+        file = file_f.read()
+
+    if verify_signature(key, sig, file):
+        logger.debug("%s passed signature verification", file_description.capitalize())
+    else:
+        raise Exception(f"{file_description.capitalize()} signature verification failed comparing {file_description} ({filename}) against sig_file ({sig_file})")
+
+
+def verify_signature(key, sig, file):
+    """
+       Verify the file signature (sig) using a public key (key)
+       with the file (file).
+    """
+
+    # Inspect the public key to determine what kind of key it is.
+    key_header = key.decode('utf-8').split('\n')[0].strip()
+
+    # PGP
+    if key_header == "-----BEGIN PGP PUBLIC KEY BLOCK-----":
+        gpg = gnupg.GPG()
+        logger.debug("Importing GPG key")
+        gpg_imported = gpg.import_keys(key.decode("utf-8"))
+        if gpg_imported.count == 1: # pylint: disable=E1101
+            logger.debug("GPG key successfully imported")
+        else:
+            raise Exception(f"Unable to import GPG key")
+
+        # The Python PGP library won't let you read a signature from memory, hence this hack.
+        with open('tmp.sig', 'wb') as sig_temp:
```

Comment:
Use a `NamedTemporaryFile` for this.
I implemented a similar hack here for the ssl library (It uses a TemporaryDirectory because i couldn't just load it and immediately remove it. In this case `NamedTemporaryFile` should work):
https://github.com/keylime/keylime/blob/9a03bd405a363546ace05c843ce25cab7f08faa6/keylime/requests_client.py#L33-L38

Reply:
Thanks for the suggestion @THS-on - I've replaced my hack with `NamedTemporaryFile`!